File: core/src/main/java/com/larryhsiao/nyx/core/jots/filter/ConstFilter.java
package com.larryhsiao.nyx.core.jots.filter;

/**
 * Constant object of {@link Filter}.
 */
public class ConstFilter implements Filter {
    private final long[] dateRange;
    private final String keyword;
    private final long[] ids;

    public ConstFilter() {
        this(new long[]{0L, 0L}, "", new long[0]);
    }

    public ConstFilter(long[] dateRange, String keyword, long[] ids) {
        this.dateRange = dateRange;
        this.keyword = keyword;
        this.ids = ids;
    }

    @Override
    public long[] dateRange() {
        return dateRange;
    }

    @Override
    public String keyword() {
        return keyword;
    }

    @Override
    public long[] ids() {
        return ids;
    }
}

File: core/src/main/java/com/larryhsiao/nyx/core/jots/filter/ConstFilter.kt
package com.larryhsiao.nyx.core.jots.filter

/**
 * Constant object of [Filter].
 */
open class ConstFilter @JvmOverloads constructor(
    private val dateRange: LongArray = longArrayOf(0L, 0L),
    private val keyword: String = "",
    private val ids: LongArray = LongArray(0)
) : Filter {
    override fun dateRange(): LongArray? {
        return dateRange
    }

    override fun keyword(): String? {
        return keyword
    }

    override fun ids(): LongArray? {
        return ids
    }
}